jobs/MeterInstanceJob.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const _ = require('lodash');
+const logger = require('../common/logger');
+const BaseJob = require('./BaseJob');
+const catalog = require('../common/models/catalog');
+const config = require('../common/config');
+const utils = require('../common/utils');
+const EventLogInterceptor = require('../common/EventLogInterceptor');
+/* jshint ignore:start */
+const maas = require('../data-access-layer/metering');
+/* jshint ignore:end */
+const apiServerClient = require('../data-access-layer/eventmesh').apiServerClient;
+const CONST = require('../common/constants');
+
+class MeterInstanceJob extends BaseJob {
+
+  /* jshint ignore:start */
+  static async run(job, done) {
+    try {
+      logger.info(`-> Starting MeterInstanceJob -  name: ${job.attrs.data[CONST.JOB_NAME_ATTRIB]} - with options: ${JSON.stringify(job.attrs.data)} `);
+      let events = await this.getInstanceEvents()
+      logger.info('recieved events -> ', events)
+      let meterResponse = await this.meter(events)
+      return this.runSucceeded(meterResponse, job, done)
+    } catch (err) {
+      return this.runFailed(err, {}, job, done)
+    }
+  }
+  /* jshint ignore:end */
+
+  static getInstanceEvents() {
+    const options = {
+      resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.INSTANCE,
+      resourceType: CONST.APISERVER.RESOURCE_TYPES.SFEVENT,
+      query: {
+        labelSelector: `state in (${CONST.METER_STATE.TO_BE_METERED},${CONST.METER_STATE.FAILED})`
+      }
+    };
+    return apiServerClient.getResources(options);
+  }
+
+  /* jshint ignore:start */
+  static async meter(events) {
+    try {
+      logger.info(`Number of events to be metered in this run - ${events.length}`);
+      // Adding this comment as we are transitioning to async/await
+      // Note: The below Promise is not bluebird promise
+      let promiseArray = await Promise.all(_.map(events, async (event) =>
+        await this.sendEvent(event)));
+      let successCount = promiseArray.filter(r => r === true).length;
```

Comment:
Minor comment: Use `const` instead of `let` since the reference is not being changed